python/packages/autogen-ext/src/autogen_ext/models/openai/_openai_client.py

```diff
@@ -426,6 +445,8 @@ async def create(
         if self.capabilities["json_output"] is False and json_output is True:
             raise ValueError("Model does not support JSON output")
 
+        if self.model_info["family"] == "o1":
+            messages = prepare_o1_messages(messages)
```

Comment:
I feel like we should make this configurable, since doesn't o1 support system messages? It's just tricky since it blocks requests sometimes?

Reply:
Developer messages are the new system messages: Starting with o1-2024-12-17, o1 models support developer messages rather than system

This seems to be the new standard, I can try just converting to developer message instead